### ISSUE TITLE
Improve client accessibility

### ISSUE THREAD

**Issue by Menelion** [linear/linear#543]
I'm investigating various bug tracking and project management solutions for my startup. The most crucial point for us is that the solution we choose must be accessible for blind managers and developers using screen readers.  
Linear has many features I'd like to see in my daily work (gitHub and Slack integrations, API, issue management, cycles…). However, accessibility at least on https://linear.app is far behind our expectations: many unlabeled buttons, inaccessible context menus and so on.  
I could describe it more if needed.

**Comment by pacocoursey** [comment]
Hi @Menelion, I work on the public website at Linear. We test primarily with VoiceOver but most pages are not optimized for screen readers and a11y tech.

> I could describe it more if needed.

Please do. If you have specific examples, I can fix them faster. Thanks!

**Comment by Menelion** [comment]
@pacocoursey Thanks for your reply! To be honest, VoiceOver is by far not the best screen reader choice, although I understand that you all develop on Macs. the most current tech stack among the blind is Windows + iOS (and Android in the countries where an iPhone is… well, too expensive for an average person to buy). Okay then, let's go. If you prefer, I can create separate issues for each inconsistency, prepare a document in your favorite format or do whatever you deem more convenient. I really like Linear's concept and it's very sad I cannot work with it on a daily basis.  
I created an account for evaluating, so I'm starting there.

### Main Page

* At the [main page](https://linear.app/accessmind/team/ACC/active) of active issues there are tons of unlabeled buttons. I wish I could tell you what they are, but I can't, or, if it's really needed, I'll call for sighted help. For example, I see two buttons between Search Workspace and Inbox. One button is after Import Issues, then there is Invite People twice (one time as a button, one time as plain text), and then another button. Please keep in mind that we blind people see webpages in a linear manner (pun intended), so there is no left and right for us. Then the navigation ends, and between the navigation and the main area there are two more unlabeled buttons. Then there is Active issues, two buttons, Teams, two more unlabeled buttons… and on and on it goes.
* the checkboxes that probably select issues, are not labeled.
* the issue numbers that definitely are links (because they open the corresponding pages) are seen as plain text.
* After an issue number there is an unlabeled button, then its title, then a date twice (I assume created and last modified?), then another unlabeled button.

### Issue Page

On the [issue page](https://linear.app/accessmind/issue/ACC-2/try-3-ways-to-navigate-linear-command-menu-keyboard-or-mouse):
* The issue title is not a heading. Actually, there are no headings at Linear, and they are the primary way of navigating pages by a blind person.
* The Description edit box is some kind of WYSIWYG thing, because I cannot see any source markup. For instance, there are videos and images, and I cannot navigate them in the edit mode.
* In the Activity… whatever it is (group, heading — appropriate markup lacks) there are Subscribe and Unsubscribe buttons. I see both which is very confusing. One of them should be either hidden or disabled (even better, make a toggle button that is pressed or unpressed).
* Comment and Edit are again plain text pieces.
* After Properties there are six unlabeled buttons.
* then there are two Todo buttons. When clicking the first of them, you can change status. By default you land on the edit box. What is it, a search box? An edit combo?
* When you select Done, for example, the buttons become Todo and Done which is also kind of confusing.
* the Add Label interface is also confusing, there is an L button with an inaccessible context menu, an edit box and some existing labels.
* If you accidentally press Escape from the issue page, you get redirected back to the Active Issues page, and browser Forward command doesn't work to bring you back to the issue you worked with before.

This is only two pages, I even didn't work with advanced features like cycles.

**Comment by pacocoursey** [comment]
Hey @Menelion, this issue was automatically closed due to inactivity via Linear's [GitHub sync](https://linear.app/docs/github#github-issues-sync). The "not planned" label is added by GitHub and is not accurately named.

Sorry for the delay here. I assumed your comment was about the Linear marketing website (which I lead) and not the in-app interface. We're aware our a11y support in the app itself is subpar. We've got a fair few complex UI patterns that are primarily optimized for keyboard shortcut and mouse users, not screen readers and accessibility tech.

We treat major a11y issues as bugs, and I believe many of those unlabeled buttons you mentioned have been fixed since then along with many other improvements. But some parts of the product with complex interactions (e.g. timeline) are certainly not usable with just the keyboard yet.

**Comment by pacocoursey** [comment]
We're actively testing and improving our accessibility support across the app. If you have examples of specific inaccessible experiences within Linear, please report them in our [community Slack](https://linearcustomers.slack.com/archives/C01GGHDBS21) so we can investigate further.
